src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/TrackerObjectManager.NativeAot.cs

```diff
@@ -0,0 +1,543 @@
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[3](pThis));
+        }
+
+        public static void FindTrackerTargetsCompleted(IntPtr pThis, bool walkFailed)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, bool, int>**)pThis)[4](pThis, walkFailed));
+        }
+
+        public static void ReferenceTrackingCompleted(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[5](pThis));
+        }
+
+        public static void SetReferenceTrackerHost(IntPtr pThis, IntPtr referenceTrackerHost)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, IntPtr, int>**)pThis)[6](pThis, referenceTrackerHost));
+        }
+    }
+
+    // Wrapper for IReferenceTracker
+    internal static unsafe class IReferenceTracker
+    {
+        public static void ConnectFromTrackerSource(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[3](pThis));
+        }
+
+        public static void DisconnectFromTrackerSource(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[4](pThis));
+        }
+
+        public static void FindTrackerTargets(IntPtr pThis, IntPtr findReferenceTargetsCallback)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, IntPtr, int>**)pThis)[5](pThis, findReferenceTargetsCallback));
+        }
+
+        public static void GetReferenceTrackerManager(IntPtr pThis, out IntPtr referenceTrackerManager)
+        {
+            fixed (IntPtr* ptr = &referenceTrackerManager)
+                Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, IntPtr*, int>**)pThis)[6](pThis, ptr));
+        }
+
+        public static void AddRefFromTrackerSource(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[7](pThis));
+        }
+
+        public static void ReleaseFromTrackerSource(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[8](pThis));
+        }
+
+        public static void PegFromTrackerSource(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[9](pThis));
+        }
+    }
+
+    // Callback implementation of IFindReferenceTargetsCallback
+    internal unsafe struct FindReferenceTargetsCallback
+    {
+        internal static GCHandle s_currentRootObjectHandle;
+
+        [UnmanagedCallersOnly]
+        private static unsafe int IFindReferenceTargetsCallback_QueryInterface(IntPtr pThis, Guid* guid, IntPtr* ppObject)
+        {
+            if (pThis == IntPtr.Zero)
+            {
+                return HResults.E_POINTER;
+            }
+
```

Comment:
```suggestion
```
this pointer is assumed to be non-null in COM calls. These null checks for this pointer are superfluous.